src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/config/JpaAuditConfig.kt

```diff
@@ -0,0 +1,44 @@
+package uk.gov.justice.digital.hmpps.sentenceplan.config
+
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.data.domain.AuditorAware
+import org.springframework.data.jpa.repository.config.EnableJpaAuditing
+import org.springframework.security.core.context.SecurityContextHolder
+import uk.gov.justice.digital.hmpps.sentenceplan.entity.PractitionerEntity
+import uk.gov.justice.digital.hmpps.sentenceplan.entity.PractitionerRepository
+import java.util.Optional
+
+@Configuration
+@EnableJpaAuditing(auditorAwareRef = "auditorProvider")
+class JpaAuditConfig(private val practitionerRepository: PractitionerRepository) {
+
+  val usernameAuthenticationNotAvailable = "Not set"
+
+  @Bean
+  fun auditorProvider(): AuditorAware<PractitionerEntity> {
+    return AuditorAware<PractitionerEntity> {
+      var practitionerEntity: PractitionerEntity
+
+      if (SecurityContextHolder.getContext().authentication != null) {
+        val usernameParts = SecurityContextHolder.getContext().authentication.name.split('|')
+        val uuid = usernameParts[0]
+        val username = usernameParts[1]
+
+        try {
+          practitionerEntity = practitionerRepository.findByUuid(uuid)
+        } catch (e: Exception) {
```

Comment:
`e` could be `_` as it's unused. Is catching `Exception` the right strategy?

Reply:
No, this is a mistake, I'll put the correct Exception in here

Reply:
Resolved in https://github.com/ministryofjustice/hmpps-sentence-plan/pull/81/commits/4c863cdf648ad7d6f8fde7ec905533be9a15c9cb